### PR TITLE
[Feat]: Create email templates

### LINKED ISSUE #168: [Feat]: Create Endpoint For Working With Email Templates in The System
## Description
Implement a flexible and efficient system for storing, retrieving, and managing email templates on the backend. The system should focus on scalability, performance, and ease of use.

## Objectives
- Create a versatile email template storage system
- Implement efficient retrieval mechanisms
- Ensure template flexibility with placeholder support
- Provide a clean API for frontend integration

## Tasks
- [x] Design template storage architecture
  - [x] Define template structure (including placeholders)
  - [x] Determine storage method

- [x] Implement template storage
  - [x] Create data model for email templates
  - [x] Implement database storage provider

- [x] Develop template retrieval system
  - [x] Create service for async template retrieval

- [x] Design and implement API
  - [x] Create endpoints for template CRUD operations
  - [x] Develop DTO for template data transfer

- [x] Implement placeholder management
  - [x] Develop system for defining and managing placeholders
  - [ ] Create mechanism for validating placeholder usage

- [ ] Develop comprehensive test suite
  - [ ] Create unit tests for core components
  - [ ] Implement integration tests for end-to-end workflows
  - [ ] Develop performance tests for template retrieval

## Acceptance Criteria
1. Email templates can be stored and retrieved efficiently
2. Templates include flexible placeholder system
3. API provides clean, RESTful endpoints for template management
4. Placeholder validation ensures template integrity
5. Comprehensive test suite validates system reliability and performance

## API RESPONSES
GET /api/templates/{id}
### Success Response
```json
   {
   "id": "string",
  "name": "string",
  "subject": "Welcome to Our Service, {{UserName}}!",
  "bodyTemplate": "<h1>Hello {{UserName}},</h1><p>Welcome to our service! We're excited to have you on board.</p><p>Your account was created on {{SignUpDate}}.</p><p>Best regards,<br>The {{CompanyName}} Team</p>",
  "placeholders": {
    "UserName": "",
    "SignUpDate": "",
    "CompanyName": "Amazing Corp"
  }
}
```

### PR DESCRIPTION
**Pull request by Ace2489**
# Description

**Closes #168**

# Changes proposed

## What were you told to do?
Create endpoints to enable creating and storing new email templates.

## What did you do?
- Created the endpoint for receiving templates.
- Implemented validation for received data to reduce errors and duplicates.
- Created backend infrastructure to parse and store template data in the database. 

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles match our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

### CODE REVIEW

File: src/Hng.Web/appsettings.json [hunk 1]
```diff
@@ -19,10 +19,10 @@
...
   },
   "SmtpCredentials": {
-    "Host": "",
-    "Port": "",
-    "Username": "",
-    "Password": ""
+    "Host": "host",
+    "Port": "1234",
+    "Username": "user",
+    "Password": "pass"
```
Comment (Godhanded) [inline src/Hng.Web/appsettings.json:25]: Please remove theses dummy values for these secretes

File: src/Hng.Web/appsettings.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "DefaultConnectionString": ""
   },
   "Jwt": {
-    "SecretKey": "",
+    "SecretKey": "Super-secret key for all of my applications here. I want to cry",
```
Comment (Godhanded) [inline src/Hng.Web/appsettings.json:13]: Remove dummy secrete key value

**Review by Godhanded — CHANGES_REQUESTED** [review]
Left two comments, please also check that route and request and response body match what's on the unified endpoints document

File: src/Hng.Web/appsettings.json (hunk 1 above)
Comment (Ace2489) [reply src/Hng.Web/appsettings.json]: I thought the successresponse and errorresponse would automatically do that.

File: src/Hng.Domain/Entities/EmailTemplate.cs [hunk 3]
```diff
@@ -1,7 +1,12 @@
+using System.ComponentModel.DataAnnotations;
+using Microsoft.EntityFrameworkCore;
+
 namespace Hng.Domain.Entities;
 
+[Index(nameof(Name), IsUnique = true)]
```
Comment (Godhanded) [inline src/Hng.Domain/Entities/EmailTemplate.cs:6]: How come this is here and not inside the class above name?

File: src/Hng.Application/Features/EmailTemplates/DTOs/CreateEmailTemplateDTO.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    [Required]
+    [JsonPropertyName("subject")]
+    public string Subject { get; set; }
+
+    [Required]
+    [JsonPropertyName("template_body")]
+    public string TemplateBody { get; set; }
+
+    [JsonPropertyName("placeholders")]
```
Comment (Godhanded) [inline src/Hng.Application/Features/EmailTemplates/DTOs/CreateEmailTemplateDTO.cs:20]: Confirm from unified docs if the place holder json property name should be snack_case

**Review by Godhanded — CHANGES_REQUESTED** [review]

File: src/Hng.Domain/Entities/EmailTemplate.cs (hunk 3 above)
Comment (Ace2489) [reply src/Hng.Domain/Entities/EmailTemplate.cs:6]: Index attribute takes classes as arguments.

File: src/Hng.Application/Features/EmailTemplates/DTOs/CreateEmailTemplateDTO.cs (hunk 4 above)
Comment (Ace2489) [reply src/Hng.Application/Features/EmailTemplates/DTOs/CreateEmailTemplateDTO.cs:20]: There isn't a document entry for the email templates — this is what I and my FE guy agreed upon.

**Review by Godhanded — APPROVED** [review]

**Pull request merged**